### ISSUE TITLE
AltBMPPolyInspections (Trees) - No Inspection Records Being Generated in Export

### ISSUE THREAD

**Issue by JuanitaMiles** [kcigeospatial/balt_co_ETL#29]
No Tree Inspection records are being generated within the MDE Export. I have a feeling the ETL may still  be trying to pull inspection records from TREE_INSP_REPORT in the source database, as the ETL documentation lists TREE_INSP_REPORT as the source table rather than INSP_TREE.

![image](https://user-images.githubusercontent.com/31770504/32385144-9000de12-c093-11e7-92af-d5d2e276d167.png)

Seeing as this table in Baltimore County's Source database is not being utilized, there are no records in the source and therefore no records are being generated in the MDE AltBMPPolyInspections. 

We can verify with Gerry if the ETL is indeed trying to pull from Tree Insp Report, then I can probably easily go through and figure out the Source to Target between INSP_TREE and AltBMPPolyInspections, and provide to Gerry to update.

Thoughts?

**Comment by leeensminger** [comment]
Yes sounds good.  I know Gerry is out today so you may want to just create a record in TREE_INSP_REPORT and test it quick to confirm.

**Comment by JuanitaMiles** [comment]
@gerrykelly Gerry are you able to confirm whether the ETL is pulling from INSP_TREE or TREE_INSP_REPORT in the source database to populate AltBMPPolyInspections in the MDE Export? For some reason I am not able to add any records to TREE_INSP_REPORT to test my theory above. 

The ETL should reference INSP_TREE. If it is not being referenced I can write up the source to target for you to make the updates to the ETL

**Comment by gerrykelly** [comment]
@JuanitaMiles 
You are correct that TREE_INSP_REPORT is used in the ETL and INSP_TREE is not.  I'll work on making the change.  But, if it is not obvious, please provide the field mapping so I can finish it.

**Comment by gerrykelly** [comment]
@JuanitaMiles 
I started on this an realized that I cannot use INSP_TREE (at least not the way I was using TREE_INSP_REPORT and the way I process almost every other table) because it does not have *_NUM field in the table (and there is no trigger that populates an ID)
Also there are NULL TREE_INFO_ID's while in the real data there should not be (correct?0

@leeensminger   Is not having a "_NUM" field/trigger an oversight, or am I missing something here on how to use the table in the ETL?

**Comment by JuanitaMiles** [comment]
@leeensminger 
- As Gerry has noted, INSP_TREE does not have a “*_Num” field that is normally used for creating the ALTBMP_LN_INSP_ID within the MDE export. The only  field I see that could serve as this NUM field would be TREE_INSPECTION_ID, but this is a GUID field and would have to be changed to a text field in order to work. We will need a field to input this “INSP_TREE_NUM” in order to export tree inspection records.

- INSP_TREE does not have an ALTBMP_STATUS field that records “Pass/Fail” status. There is an INSPECTION_STATUS field with a true/false domain. Can we use this field and translate the True/False domain to the Pass/Fail domain in the MDE export?

**Comment by leeensminger** [comment]
I have some thoughts let's discuss as a team on Monday

**Comment by leeensminger** [comment]
@gerrykelly per our phone call:
at the start of the Reforestation inspection strategy process, the system should:
1. delete all records from TREE_INSP_REPORT
2. take the records from INSP_TREE table, and append to TREE_INSP_REPORT.  See translation map below.
2a. I don't care when the selection set is applied - before you do this append, or after.  But if you append all, then do the selection set, you should probably delete all records from the TREE_INSP_REPORT table after the ETL processes to keep things clear for any ArcGIS Desktop users.
3. continue processing as the ETL currently is

For # 2 above:
![image](https://user-images.githubusercontent.com/4103847/33853446-fab3c45c-de8b-11e7-9394-cfb93ce3fee8.png)

**Comment by gerrykelly** [comment]
@leeensminger @BC89 
I wrote the routine to do this, but TREE_INSP_REPORT_NUM is not being populated on the row inserts.  There is no trigger.

**Comment by leeensminger** [comment]
I emailed Dustin and BC offline.  Hopefully we are still in a position with the project where the trigger can be implemented.

**Comment by leeensminger** [comment]
The ETL fails when running at KCI this morning.  This ran fine yesterday, so must be related to a recent code push.  It consistently fails at the TREE_INSP_REPORT.  I ran it once, got the error, set REPORTING_YEAR to null for all tree related strategies (TREE_INFO, TREE_AREAS, INSP_TREE, TREE_INSP_REPORT), and ran it again and still got the error.

![image](https://user-images.githubusercontent.com/4103847/33900722-7de7d0dc-df3d-11e7-8586-9c303f7f99eb.png)

**Comment by gerrykelly** [comment]
@leeensminger 
Where did we leave off on this?  Is it working here?  At the county?

**Comment by gerrykelly** [comment]
@leeensminger 
I'm going to get back to this today.  Since I saw that  Append_Failure_TREE_INSP_REPORT_to_AltBMPPolyInspections is being generated with all expected records I was gong to assign it back to you for data/schema issues, but I see that altbmp_py_id is not getting populated correctly, so I'll need to fix that then pass it back to you.

**Comment by gerrykelly** [comment]
Fixed and now append to AltBMPPolyInspections works.
The issue was that there were null tree_info_ids in insp_tree for 2017.  Also, all the others gave a value of orphan for altbmp_py_id.  Rather than leaving them null and having the append fail, I now give them a value of "missing".  This a change that applies to all tables, so we could see more successful appends, but should be aware of the "missing"s

**Comment by gerrykelly** [comment]
@leeensminger 
Also, a reminder that this only works bc I calculated dummy TREE_SITE_NUMs in tree_info.  I'll make that its own issue and assign you and @BC89 Brendan

from my 12/14 email:
In trying to add the tree info back to AltBMPPoly I set the reporting years (which were all null) in tree_info & insp_tree to 2017.  So while NPDES.TREE_INSP_REPORT now gets populated and its new insert trigger works, the ETL is failing bc all the TREE_SITE_NUM in tree_info are null.  There is no insert trigger for TREE_info.

I’ll go ahead and calc dummy values and retry but we need a trigger.
